planet/clients/subscriptions.py

```diff
@@ -46,22 +46,17 @@ async def list_subscriptions(self,
             ClientError
 
         """
-        # Temporary marker for behavior of module with unpatched state.
-        if _fake_subs is None:
-            raise NotImplementedError
-
-        if status:
+        try:
             select_subs = (dict(**sub, id=sub_id) for sub_id,
                            sub in _fake_subs.items()
-                           if sub['status'] in status)
-        else:
-            select_subs = (
-                dict(**sub, id=sub_id) for sub_id, sub in _fake_subs.items())
+                           if not status or sub['status'] in status)
+            filtered_subs = itertools.islice(select_subs, limit)
 
-        filtered_subs = itertools.islice(select_subs, limit)
+            for sub in filtered_subs:
+                yield sub
 
-        for sub in filtered_subs:
-            yield sub
+        except Exception as server_error:
+            raise ClientError("Subscription failure") from server_error
```

Comment:
We discussed and ultimately decided against exception chaining. While we can revisit in a larger review, for now let's change this to match how exceptions are raised from the other clients. Ref: https://docs.google.com/document/d/1tdwziX0U9Tsz49G6_qJKr3YgMATQGIx0pHsxpTC7Pf8/edit

Reply:
I would really want this if I were a user of the Python client, and in other open source projects I often see contributors doing exceptions like this because they want it too. I'm going to leave it like this to help me during development -- it's useful! -- and then if we really want to truncate the exceptions, let's do it before a pre-release.

Reply:
OK, so I have to admit I am not fully understanding what is 'mock code' and what is proposed production code here. For production code, the handling of server-side exceptions should follow entry 2b in the above-linked SDK Error Design doc. That is, they are raised, unaltered. ClientErrors originate from client-side issues, and, following 2b, they are not chained.

Reply:
Just found where this is documented in the code (making a ticket to add this to the docs) - https://github.com/planetlabs/planet-client-python/blob/main/DESIGN.md#errors